crypto/signer.go

```diff
@@ -23,31 +23,33 @@ import (
 	"github.com/google/trillian/crypto/sigpb"
 )
 
-var (
-	signerHashLookup = map[sigpb.DigitallySigned_HashAlgorithm]crypto.Hash{
-		sigpb.DigitallySigned_SHA256: crypto.SHA256,
-	}
-	reverseSignerHashLookup = map[crypto.Hash]sigpb.DigitallySigned_HashAlgorithm{
-		crypto.SHA256: sigpb.DigitallySigned_SHA256,
-	}
-)
+var sigpbHashLookup = map[crypto.Hash]sigpb.DigitallySigned_HashAlgorithm{
+	crypto.SHA256: sigpb.DigitallySigned_SHA256,
+}
 
 // Signer is responsible for signing log-related data and producing the appropriate
 // application specific signature objects.
 type Signer struct {
-	hash         crypto.Hash
-	signer       crypto.Signer
-	sigAlgorithm sigpb.DigitallySigned_SignatureAlgorithm
+	hash crypto.Hash
+	key  PrivateKeyManager
 }
 
 // NewSigner creates a new Signer wrapping up a hasher and a signer. For the moment
 // we only support SHA256 hashing and either ECDSA or RSA signing but this is not enforced
 // here.
 func NewSigner(sigAlgo sigpb.DigitallySigned_SignatureAlgorithm, signer crypto.Signer) *Signer {
+	return NewSignerFromPrivateKeyManager(localSigner{
```

Comment:
Almost there. 

Consider making `NewSignerFromKeyManager` call `NewSigner` rather than the other way around.  Observe that the current state requires creating an extra intermediate object.